bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -457,4 +491,82 @@ public static boolean isValidIPConfig(String ipAddress) {
         return false;
     }
 
+    private void scheduleToPollNetworkInterface(Map<String, Object> props) {
+
+        final NetUtil myself = this;
+        if (networkInterfacePollFuture != null) {
+            networkInterfacePollFuture.cancel(true);
+            networkInterfacePollFuture = null;
+        }
+
+        Runnable tempRunnable = new Runnable() {
+            @Override
+            public void run() {
+                myself.pollAndNotifyNetworkInterfaceAddress();
+            }
+        };
+
+        String pollFrequenceSecondsStr = "";
+        int pollFrequenceSeconds = INTERFACE_POLL_DEFAULT_FREQUENCY_SECONDS;
+        try {
+            pollFrequenceSecondsStr = (String) props.get(INTERFACE_POLL_FREQUENCY_SECONDS_KEY);
+            pollFrequenceSeconds = Integer.parseInt(pollFrequenceSecondsStr);
+        } catch (NumberFormatException e) {
+            LOGGER.debug("Cannot parse value {} from key {}, will use default {}", pollFrequenceSecondsStr,
+                    INTERFACE_POLL_FREQUENCY_SECONDS_KEY, INTERFACE_POLL_DEFAULT_FREQUENCY_SECONDS);
+        }
+        networkInterfacePollFuture = (ScheduledFuture<Void>) scheduledExecutorService
```

Comment:
Why do you need a cast here?

Reply:
ScheduledFuture<Void> was declared, instead of ScheduledFuture.  Its removed.

Reply:
Removed totally to avoid generics.